src/styles/main.scss

```diff
@@ -1,3 +1,71 @@
-body {
-  margin: 0;
+@import "./utils/variables.scss";
+
+.page {
+  position: relative;
+  height: 100vh;
+}
+
+.calendar {
+  border: 1px solid black;
+
+  height: 505px;
+  width: 707px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  padding: 10px;
+
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+
+  &__days {
+    position: relative;
+    box-sizing: border-box;
+    height: 100px;
+    width: 100px;
+    background-color: #eee;
+    border: 1px solid black;
+
+    cursor: pointer;
+    transition-duration: 500ms;
+    transition-property: background-color, transform;
+
+    &:hover {
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  &__days::before {
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    margin-right: -50%;
+    transform: translate(-50%, -50%);
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__days:nth-child(#{$i}) {
+    &::before {
+      content: "#{$i}";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+}
+
+@each $day, $n in $days {
+  .calendar--start-day--#{$day} > .calendar__days:first-child {
+    margin-left: (($ceil-width + $gap-size) * $n);
+  }
+}
+
+@for $length from 28 through 31 {
+  .calendar--month-length--#{$length}
+  > .calendar__days:nth-child(n + #{$length + 1}) {
+    display: none;
+  }
 }
```

Comment:
as i see it already working, but would be better to use nesting of SCSS and move these func to .calendar or .calendar__day